src/UsgsAstroFramePlugin.cpp

```diff
@@ -273,18 +278,55 @@ return sensor_model;
 }
 
 
-csm::Model *UsgsAstroFramePlugin::constructModelFromISD(const csm::Isd &imageSupportData,
+csm::Isd UsgsAstroFramePlugin::loadImageSupportData(const csm::Isd &imageSupportDataOriginal) const{
+  // Get image location from the input csm::Isd: 
+  std::string imageFilename = imageSupportDataOriginal.filename(); 
+  
+  // Load 'sidecar' ISD file
+  size_t lastIndex = imageFilename.find_last_of("."); 
+  std::string baseName = imageFilename.substr(0, lastIndex); 
+  std::string isdFilename = baseName.append(".json");
+
+  csm::Isd imageSupportData(isdFilename);
+  imageSupportData.clearAllParams();
+
+  std::ifstream isdFile(isdFilename); 
```

Comment:
We should put a try/catch around this file IO in case the file doesn't exist or there's an issue with it. I don't think the ifstream error is going to be descriptive enough.